### PR TITLE
2.13.6

### PR DESCRIPTION
**Pull request by CiprianDraghici**

### CODE REVIEW

**Review by MiroMargineanu — APPROVED** [review]

File: src/hooks/transactions/helpers/setTransactionNonces.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  if (transactions.length === 0) {
+    return transactions;
+  }
+
+  return transactions.map((tx: Transaction, index: number) => {
+    const nextNonce = latestNonce + index;
+
+    const txNonce = tx.getNonce().valueOf();
+    // stop replacing nonce if transaction is configured with a higher nonce than the existing one
+    const computedNonce = txNonce > nextNonce ? txNonce : nextNonce;
```
Comment (arhtudormorar) [inline src/hooks/transactions/helpers/setTransactionNonces.ts:16]: just a suggestion: maybe create a small helper called `getTransactionNonce` and use it here and in newTransaction so we isolate this logic in one place

File: src/hooks/transactions/helpers/setTransactionNonces.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  transactions: Array<Transaction>
+): Array<Transaction> => {
+  if (transactions.length === 0) {
+    return transactions;
+  }
+
+  return transactions.map((tx: Transaction, index: number) => {
+    const nextNonce = latestNonce + index;
+
+    const txNonce = tx.getNonce().valueOf();
```
Comment (arhtudormorar) [inline src/hooks/transactions/helpers/setTransactionNonces.ts:15]: transactionNonce

File: src/hooks/transactions/helpers/setTransactionNonces.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { Transaction } from '@multiversx/sdk-core';
+import { computeTransactionNonce } from 'services/transactions/computeTransactionNonce';
+
+export const setTransactionNonces = (
+  latestNonce: number,
```
Comment (arhtudormorar) [inline src/hooks/transactions/helpers/setTransactionNonces.ts:5]: aici as merge pe named parameters cu tip mai sus

File: src/services/transactions/transformAndSignTransactions.ts [hunk 4]
```diff
@@ -42,9 +56,13 @@ export async function transformAndSignTransactions({
...
       gasPrice = GAS_PRICE,
-      gasLimit = calculateGasLimit(tx.data),
+      gasLimit = calculateGasLimit({
+        data: tx.data,
+        isGuarded: account?.isGuarded
+      }),
       guardian,
-      guardianSignature
+      guardianSignature,
+      nonce: txNonce = 0
```
Comment (arhtudormorar) [inline src/services/transactions/transformAndSignTransactions.ts:65]: transactionNonce rename

File: src/services/transactions/transformAndSignTransactions.ts [hunk 5]
```diff
@@ -55,6 +73,11 @@ export async function transformAndSignTransactions({
       throw ErrorCodesEnum.invalidReceiver;
     }
 
+    const computedNonce = computeTransactionNonce({
+      accountNonce: nonce,
```
Comment (arhtudormorar) [inline src/services/transactions/transformAndSignTransactions.ts:77]: la linia 49 direct accountNonce

**Review by MiroMargineanu — APPROVED** [review]

**Review by arhtudormorar — APPROVED** [review]

**Review by MiroMargineanu — APPROVED** [review]

**Pull request merged**